### PR TITLE
Add deploy pipeline documentation

### PR DESCRIPTION
**Pull request by hakubo**

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -33,6 +28,61 @@ In order to run service locally you need to configure hosts. Add below line to `
 127.0.0.1	wikia-local.com dev.wikia-local.com muppet.dev.wikia-local.com
 ```
 
+## Deploy Pipeline
+
+This section describe how to depoly mediawiki to various environments, how they work and how to introduce changes to it.
+
+There are 2 main parts of all deployments:
+1. Jenkins job that is defined in jenkins-jobs reposiotory that accepts needed parameters to deploy to a given environment
+2. a jenkins file that is in this repo that is being executed by Jenkins as a pipeline job
```
Comment (nikodamn) [inline docker/README.md:37]: "a `Jenkinsfile` that is..." would be easier to found

File: docker/README.md [hunk 2]
```diff
@@ -33,6 +28,61 @@ In order to run service locally you need to configure hosts. Add below line to `
 127.0.0.1	wikia-local.com dev.wikia-local.com muppet.dev.wikia-local.com
 ```
 
+## Deploy Pipeline
+
+This section describe how to depoly mediawiki to various environments, how they work and how to introduce changes to it.
```
Comment (nikodamn) [inline docker/README.md]: `deploy` typo

File: docker/README.md [hunk 3]
```diff
@@ -33,6 +28,61 @@ In order to run service locally you need to configure hosts. Add below line to `
...
+
+There are 2 main parts of all deployments:
+1. Jenkins job that is defined in jenkins-jobs reposiotory that accepts needed parameters to deploy to a given environment
+2. a jenkins file that is in this repo that is being executed by Jenkins as a pipeline job
+
+### How to deploy to sandbox
+
+1. Open [mediaiwki-deploy-sandbox](http://jenkins.wikia-prod:8080/blue/organizations/jenkins/mediawiki-deploy-sandbox/activity)
```
Comment (nikodamn) [inline docker/README.md]: `mediawiki` typo

File: docker/README.md (hunk 1 above)
Comment (nikodamn) [reply docker/README.md:37]: Ok, I see it in the line #46 now. Nevermind

File: docker/README.md [hunk 4]
```diff
@@ -33,6 +28,61 @@ In order to run service locally you need to configure hosts. Add below line to `
...
+
+#### Backstage
+- jenkins file: https://github.com/Wikia/app/blob/dev/docker/sandbox/Jenkinsfile
+
+
+### How to deploy to preview
+
+This job is ran by [daily-release-cut](http://jenkins.wikia-prod:8080/blue/organizations/jenkins/daily-release-cut/activity) but if you need to deploy manually:
+
+1. Open [mediaiwki-deploy-preview](http://jenkins.wikia-prod:8080/blue/organizations/jenkins/mediawiki-deploy-preview/activity)
```
Comment (nikodamn) [inline docker/README.md]: What do you think about adding warning e.g. "Please keep in my that preview deployment will automatically start all functional tests" or similar?

File: docker/README.md [hunk 5]
```diff
@@ -33,6 +28,61 @@ In order to run service locally you need to configure hosts. Add below line to `
...
+2. Press run button, fill in the form
+	1. new release - will create new release branch, new release tag and push that to preview
+	2. hotfix - will chekout a release branch that is currently on preview and if there are any changes on this branch against the tag that is deployed - it will create new tag and push that to preview
```
Comment (nikodamn) [inline docker/README.md]: `checkout` typo

File: docker/README.md [hunk 6]
```diff
@@ -33,6 +28,61 @@ In order to run service locally you need to configure hosts. Add below line to `
...
+	3. latest - fetch a latest tag that is on github and push that to preview
+3. Submit the form
+4. This job will wait for your input on stage 'Fetch version' - if you won't press Proceed it will be aborted
+
+### how to deploy to prod
+1. Open [mediaiwki-deploy-prod](http://jenkins.wikia-prod:8080/blue/organizations/jenkins/mediawiki-deploy-prod/activity)
```
Comment (nikodamn) [inline docker/README.md]: `mediawiki` typo

**Review by nikodamn — APPROVED** [review]

**Review by pwojcik86 — APPROVED** [review]

File: docker/README.md [hunk 7]
```diff
@@ -33,6 +28,63 @@ In order to run service locally you need to configure hosts. Add below line to `
 127.0.0.1	wikia-local.com dev.wikia-local.com muppet.dev.wikia-local.com
 ```
 
+## Deploy Pipeline
+
+This section describe how to deploy mediawiki to various environments, how they work and how to introduce changes to it.
+
+There are 2 main parts of all deployments:
+1. Jenkins job that is defined in jenkins-jobs reposiotory that accepts needed parameters to deploy to a given environment
```
Comment (KockaAdmiralac) [inline docker/README.md]: minor: `reposiotory` -> `repository`

**Review by macbre — APPROVED** [review]

File: docker/README.md [hunk 8]
```diff
@@ -33,6 +28,63 @@ In order to run service locally you need to configure hosts. Add below line to `
...
+
+There are 2 main parts of all deployments:
+1. Jenkins job that is defined in jenkins-jobs repository that accepts needed parameters to deploy to a given environment
+2. a jenkins file that is in this repo that is being executed by Jenkins as a pipeline job
+
+### How to deploy to sandbox
+
+1. Open [mediawiki-deploy-sandbox](http://jenkins.wikia-prod:8080/blue/organizations/jenkins/mediawiki-deploy-sandbox/activity)
+2. Press run button, fill in the form and submit the form
+3. Profit!
```
Comment (macbre) [inline docker/README.md:43]: You can mention Slack integration.

**Pull request merged**